src/client/Microsoft.Identity.Client/Platforms/Features/OpenTelemetry/OtelInstrumentation.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Cache;
+using Microsoft.Identity.Client.TelemetryCore;
+using Microsoft.Identity.Client.TelemetryCore.OpenTelemetry;
+using System.Diagnostics.Metrics;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.Platforms.Features.OpenTelemetry
+{
+    /// <summary>
+    /// Class to hold the OpenTelemetry objects used by MSAL.
+    /// </summary>
+    internal class OtelInstrumentation : IOtelInstrumentation
+    {
+        /// <summary>
+        /// Constant to hold the name of the Meter.
+        /// </summary>
+        public const string MeterName = "MicrosoftIdentityClient_Common_Meter";
+
+        /// <summary>
+        /// Constant to holt the name of the ActivitySource.
+        /// </summary>
+        public const string ActivitySourceName = "MicrosoftIdentityClient_Activity";
+
+        private const string SuccessCounterName = "MsalSuccess";
+        private const string FailedCounterName = "MsalFailed";
+        private const string TotalDurationHistogramName = "MsalTotalDuration.1A";
```

Comment:
Could these be enums instead?

Reply:
No, these are strings for names of counters and histograms. Does not make sense as enum.